pkg/auth/oauth/handlers/interfaces.go

```diff
@@ -25,6 +26,50 @@ type AuthenticationRedirector interface {
 	AuthenticationRedirect(w http.ResponseWriter, req *http.Request) (err error)
 }
 
+// AuthenticationRedirectors is a collection that provides a list of
+// acceptable authentication mechanisms
+type AuthenticationRedirectors struct {
+	Count         int
```

Comment:
It's not strictly necessary, but having that available made several other checks much less ugly (since otherwise I had to do len(ar.Names))